src/components/PeopleTable.tsx

```diff
@@ -1,4 +1,36 @@
-export const PeopleTable = () => {
+import { useMemo } from 'react';
+import { useSearchParams } from 'react-router-dom';
+import classNames from 'classnames';
+import { Person } from '../types';
+import { PersonProperties } from './PersonProperties';
+import { getFiltredPeople } from '../utils/getFiltredPeople';
+import { SearchLink } from './SearchLink';
+import { getSortedPeople } from '../utils/getSortedPeople';
+
+type Props = {
+  people: Person[],
+  selectPeopleSlug: string | null,
+};
+
+export const PeopleTable: React.FC<Props> = ({
+  people,
+  selectPeopleSlug,
+}) => {
+  const [searchParams] = useSearchParams();
+  const query = searchParams.get('query') || '';
+  const sex = searchParams.get('sex') || '';
+  const centuries = searchParams.getAll('centuries');
+  const sort = searchParams.get('sort');
+  const order = searchParams.get('order');
+
+  const filteredPeople = useMemo(() => {
+    return getFiltredPeople(people, searchParams);
+  }, [people, query, sex, centuries]);
```

Comment:
Do you really need to add to the dependencies `query`, `sex` and `centuries`? You can just pass `searchParams` instead of these all variables and result will be the same.